src/ipa_pc/mod.rs

```diff
@@ -382,104 +382,125 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
+            // LC(C) = SUM ( lambda^i * z_i(x) * C_i' )
+            for (index, &comm_dash_segment) in comm_dash.iter().enumerate() {
+                if index == lc_comm.len() {
+                    lc_comm.push(G::zero());
+                }
+                lc_comm[index] = (lc_comm[index].into_projective()
+                    + &(comm_dash_segment.into_projective()
+                        .mul(&z_i_value)
+                        .mul(&cur_challenge)
+                    )
+                ).into_affine();
+            }
 
+            // lambda^i
             cur_challenge = cur_challenge * &lambda;
         }
 
-        let mut batch_values = vec![];
-        for labeled_commitment in commitments.iter() {
-            batch_values.push(batch_proof.batch_values[
-                *labels
-                    .get(labeled_commitment.label())
-                    .ok_or(Error::MissingEvaluation {
-                        label: labeled_commitment.label().to_string(),
-                    })?
-                ]
-            );
-        }
+        // z(x)
+        let z_x_value = z_polynomial.evaluate(x_point);
 
-        // Reconstructed v value added to the check
-        batch_values.push(computed_batch_v);
+        // LC(C) = SUM ( lambda^i * z_i(x) * C_i' ) - z(X) * C
+        lc_comm = lc_comm.iter().enumerate().map(|(index, &lc_comm_segment)| {
+            let h_comm_segment = if batch_proof.h_comm.len() > index {
+                batch_proof.h_comm[index]
+            } else {
+                G::zero()
+            };
+            (lc_comm_segment.into_projective() + &h_comm_segment.mul(-z_x_value)).into_affine()
+        }).collect::<Vec<_>>();
 
-        // The commitment to h(X) polynomial added to the check
-        let mut commitments = commitments;
+        // Expected open value added to the check
+        let batch_values = vec![G::ScalarField::zero()];
+
+        // The reconstructed commitment to h(X) polynomial put to the check
         let labeled_batch_commitment = LabeledCommitment::new(
-            format!("Batch"),
-            Commitment { comm: batch_commitment.clone(), shifted_comm: None },
+            format!("LC"),
+            Commitment { comm: lc_comm.clone(), shifted_comm: None },
             None
         );
-        commitments.push(&labeled_batch_commitment);
+        let batch_commitments = vec![&labeled_batch_commitment]; // commitments;
 
         let proof = &batch_proof.proof;
         end_timer!(evals_time);
 
         let check_time = start_timer!(|| "Succinct check batched polynomial");
 
-        fs_rng.absorb(&to_bytes![batch_proof.batch_values].unwrap());
-
-        let check_poly = Self::succinct_check(vk, commitments, point, batch_values, proof, fs_rng)?;
+        let check_poly = Self::succinct_check(
+            vk,
+            batch_commitments,
+            x_point,
+            batch_values,
+            proof,
+            fs_rng
+        )?;
```

Comment:
Here, we hand over the proof to the single-point single-poly opening proof verifier, but we did not check if it carries the expected opening value 0!

Reply:
Ah, my mistake.You have set the batch_values to the correct value.